src/policy/satisfy.rs

```diff
@@ -112,48 +112,30 @@ impl<Pk: MiniscriptKey + PublicKey32 + ToPublicKey> Policy<Pk> {
             Fragment::Sha256(hash) => satisfier
                 .lookup_sha256(hash)
                 .map(|preimage| self.compile_witness_bytes(preimage.as_ref())),
-            Fragment::And(sub_policies) => {
-                assert_eq!(
-                    2,
-                    sub_policies.len(),
-                    "Conjunctions must have exactly two sub-policies"
-                );
-
-                let mut left = sub_policies[0].satisfy_helper(satisfier)?;
-                let right = sub_policies[1].satisfy_helper(satisfier)?;
+            Fragment::And { left, right } => {
+                let mut left = left.satisfy_helper(satisfier)?;
+                let right = right.satisfy_helper(satisfier)?;
                 left.witness.extend(right.witness);
                 left.witness_cost += right.witness_cost;
                 left.program = compiler::and(left.program, right.program).ok()?;
                 left.program_cost += right.program_cost + 1;
 
                 Some(left)
             }
-            Fragment::Or(sub_policies) => {
-                assert_eq!(
-                    2,
-                    sub_policies.len(),
-                    "Disjunctions must have exactly two sub-policies"
-                );
-
+            Fragment::Or { left, right } => {
                 // TODO: Replace std::u64::MAX in MSRV >=1.43.0
```

Comment:
Let's fix these TODOs as well: https://gist.github.com/uncomputable/68e1b3c8842c9d4a38f2ef4d97219293

Reply:
Lol, yeah, I'll add a commit here.

I'm sure I fixed these *somewhere* but I guess it didn't make it into this PR.